baselines/dom.generated.d.ts

```diff
@@ -14190,6 +14209,66 @@ declare var TreeWalker: {
     new(): TreeWalker;
 };
 
+interface TrustedHTML {
+    toJSON(): string;
+    toString(): string;
+}
```

Comment:
Can you give me some more insight into TrustedHTML nodes? Is it really this simple of an interface, or is there something "more" to them?

The reason I ask is that in a structural language like TypeScript, you could accidentally assign lots of stuff to `TrustedHTML`. Is there some internal runtime bit set, or is this really anything with a `toJSON` and a `toString`?

Reply:
`TrustedHTML` is interesting, in that it's both an interface and also the name of a global object.

So it's sort of like a class (you can do `value instanceof TrustedHTML`), but the main point of the API is that it *can't* be constructed without the Trusted Types API. Sort of like you have to do `document.createElement("div")` instead of `new HTMLDivElement()`.

Each TrustedHTML instance is also associated with a specific string policy. So conceptually it might make sense to have something like `TrustedHTML<"dompurify">`. I don't know if it makes sense to include that in TypeScript, though. The `"dompurify"` policy in that case is something that would be enforced by CSP for e.g. `innerHTML` assignment, which seems a bit outside the scope of type checking.

https://developer.mozilla.org/en-US/docs/Web/API/TrustedHTML#methods

Reply:
Maybe consider emitting a class with a `private` constructor instead of an interface with these very-widespread-methods then? That'd be allowable in `instanceof` and a private member should enable nominal checking (and thus forbid structural matches).

Reply:
How would that interact with something like https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60417 ?

Element classes like `HTMLDivElement` are declared using the same `interface`+`var` pattern as this PR. Is there a particular reason to diverge from that entirely?

For what it's worth, `new HTMLDivElement()` also doesn't error in TypeScript today.